lib/ansible/modules/cloud/cloudstack/cs_rootpassword.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2018, Gregor Riepl <onitake@gmail.com>
+# based on cs_sshkeypair (c) 2015, René Moser <mail@renemoser.net>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: cs_rootpassword
+short_description: Allows resetting VM root passwords on Apache CloudStack based clouds.
+description:
+    - Reset the root password on a virtual machine.
+    - Requires cloud-init installed in the virtual machine.
+version_added: '2.8'
+author: "Gregor Riepl (@onitake)"
+options:
+  vm:
+    description:
+      - Name of the virtual machine to reset the root password on.
+    required: true
+  domain:
+    description:
+      - Name of the domain the virtual machine belongs to.
+    required: false
+    default: null
+  account:
+    description:
+      - Account the virtual machine belongs to.
+    required: false
+    default: null
+  project:
+    description:
+      - Name of the project the virtual machine belongs to.
+    required: false
+    default: null
+  poll_async:
+    description:
+      - Poll async jobs until job has finished.
+    default: yes
+    type: bool
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
+# reset and get the root password
+- local_action: cs_rootpassword name=myvirtualmachine
+  register: root
+- debug: msg='new root password is {{ root.password }}'
+# reboot the virtual machine to activate the new password
+- local_action: cs_instance name=myvirtualmachine state=restarted
+  when: root is changed
+'''
+
+RETURN = '''
+---
+id:
+  description: ID of the virtual machine.
+  returned: success
+  type: string
+  sample: a6f7a5fc-43f8-11e5-a151-feff819cdc9f
+password:
+  description: The new root password.
+  returned: success
+  type: string
+  sample: ahQu5nuNge3keesh
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    CloudStackException,
+    cs_required_together,
+    cs_argument_spec
+)
+
+class AnsibleCloudStackPasswordReset(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(AnsibleCloudStackPasswordReset, self).__init__(module)
+        self.returns = {
+            'password':   'password',
```

Comment:
minor too foo spaces.

Reply:
Not sure why there are 'too few' spaces, I'd rather remove more. But I aligned it with the code below, so should be ok now.